dbt_platform_helper/providers/io.py

```diff
@@ -1,18 +1,21 @@
-import os
-
 import click
 
 from dbt_platform_helper.platform_exception import PlatformException
-
-DEBUG = os.environ.get("DEBUG", False)
+from dbt_platform_helper.providers.environment_variable import (
+    EnvironmentVariableProvider,
+)
 
 
 class ClickIOProvider:
+    def __init__(self, env_provider=None):
+        self.env_provider = env_provider or EnvironmentVariableProvider()
+
     def warn(self, message: str):
         click.secho(message, fg="magenta")
 
     def debug(self, message: str):
-        if DEBUG == "True":
+        debug_value = self.env_provider.get("DEBUG")
+        if debug_value and debug_value.upper() == "TRUE":
```

Comment:
Looks like you've changed behaviour to make this env var a little more forgiving which is good. Can you add some test cases in for this? Looks like there's no tests currently, so if you can test the various cases that would be great.
- DEBUG not set
- DEBUG ""
- DEBUG "True"
- DEBUG "true"
- DEBUG "TrUe"
- message empty
That's probably it unless you can think of any others?